### PR TITLE
Overriding queue name to generate an array of queue names

### PR DESCRIPTION
**Pull request by juliancheal**
this allows us to create an array of child managers for our queue.

### CODE REVIEW

**Comment by juliancheal** [comment]
This PR relies on https://github.com/ManageIQ/manageiq/pull/15711

**Comment by juliancheal** [comment]
@miq-bot remove_label wip

File: app/models/manageiq/providers/amazon/cloud_manager/refresh_worker.rb [hunk 1]
```diff
@@ -1,3 +1,16 @@
 class ManageIQ::Providers::Amazon::CloudManager::RefreshWorker < ManageIQ::Providers::BaseManager::RefreshWorker
   require_nested :Runner
+
+  # overriding queue_name_for_ems so PerEmsWorkerMixin picks up *all* of the
+  # Amazon-manager types from here.
+  # This way, the refresher for Amazon's CloudManager will refresh *all*
+  # of the Amazon inventory across all managers.
+  def self.queue_name_for_ems(ems)
+    return ems unless ems.kind_of?(ExtManagementSystem)
+    ["ems_#{ems.id}"] + ems.child_managers.collect { |manager| "ems_#{manager.id}" }
```
Comment (blomquisg) [inline app/models/manageiq/providers/amazon/cloud_manager/refresh_worker.rb]: The parent implementation handles strange edge cases like if the input is a `Host`.

Maybe this should be:

```ruby
if ems.kind_of?(ExtManagementSystem)
  ["ems_#{ems.id}"] + ems.child_managers.collect { |manager| "ems_#{manager.id}" }
else
  super
end
```

Then, let `super` handle any of the edge cases.

Comment (juliancheal) [inline app/models/manageiq/providers/amazon/cloud_manager/refresh_worker.rb]: Yeah agreed. I think I was too happy to have found my last bug!

**Comment by juliancheal** [comment]
@durandom @blomquisg merge?

**Comment by durandom** [comment]
Dont we need the dependent PR merged first?

**Comment by juliancheal** [comment]
> Dont we need the dependent PR merged first?

To be honest, I don't know which PR needs merging first :(

**Pull request merged**